wgpu-core/src/device/life.rs

```diff
@@ -786,7 +787,9 @@ impl<A: HalApi> LifetimeTracker<A> {
                         if let Some(lay) =
                             hub.bind_group_layouts.unregister_locked(id.0, &mut *guard)
                         {
-                            self.free_resources.bind_group_layouts.push(lay.raw);
+                            if let BglOrDuplicate::Inner(inner) = lay.inner {
+                                self.free_resources.bind_group_layouts.push(inner.raw);
+                            }
```

Comment:
nitpick:

```suggestion
                            if let Some(inner) = lay.into_inner() {
                                self.free_resources.bind_group_layouts.push(inner.raw);
                            }
```

Pushed as a `fixup!` with 66dcca3d.

Reply:
note: This is the first time I've seen this chain-following code for potentially duplicated BGLs. Seems sound to me, after taking some time with it. 👍🏻